### PR TITLE
dyn_env: drop tests using old dev env

### PR DESCRIPTION
**Pull request by evgeniiz321**
Changes:
1. Old pytest_tests removed
2. Common code from old pytest_tests and robot directories moved closely to new tests 
3. Corresponding imports updated  
4. dynamic_env_pytest_tests renamed just to pytest_tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,14 +32,14 @@ activate = 1
...
 
 
-7. Prepare virtualenv
+2. Prepare and activate virtualenv
 
 ```shell
-$ make venv.local-pytest
-$ . venv.local-pytest/bin/activate
+$ make venv.no-dev-env-pytest
+$ . venv.no-dev-env-pytest/bin/activate
```
Comment (roman-khimov) [inline README.md:39]: Naming can be simplified in future.

File: README.md [hunk 2]
```diff
@@ -117,15 +69,19 @@ If none of the options worked for you, please complete the instruction with your
 
 ### Run and get report
 
-1. Run tests
+1. Binaries
+
+By default binaries are downloaded automatically by tests, but if you place binaries under current directory, 
+they will be taken from there.
```
Comment (roman-khimov) [inline README.md:75]: Binary list should be provided.

File: pytest_tests/lib/helpers/http_gate.py [hunk 3]
```diff
@@ -244,10 +241,11 @@ def upload_via_http_gate_curl(
     try:
         response_json = json.loads(output)
     except json.JSONDecodeError:
-        raise AssertionError(f'Invalid JSON response: {output}')
-    if 'object_id' not in response_json:
+        raise AssertionError(f"Invalid JSON response: {output}")
+    if "object_id" not in response_json:
```
Comment (roman-khimov) [inline pytest_tests/lib/helpers/http_gate.py:245]: Why these changes?

File: pytest_tests/tests/services/s3_gate/test_s3_bucket.py [hunk 4]
```diff
@@ -27,15 +27,23 @@ class TestS3GateBucket(TestNeofsS3GateBase):
...
             acl = "private"
-            bucket = s3_gate_bucket.create_bucket_s3(self.s3_client, object_lock_enabled_for_bucket=True, acl=acl, bucket_configuration="rep-1")
-            bucket_acl = s3_gate_bucket.get_bucket_acl(self.s3_client, bucket)
-            assert_bucket_s3_acl(
-                acl_grants=bucket_acl, permitted_users="CanonicalUser", acl=acl
+            bucket = s3_gate_bucket.create_bucket_s3(
+                self.s3_client,
+                object_lock_enabled_for_bucket=True,
+                acl=acl,
+                bucket_configuration="rep-1",
```
Comment (roman-khimov) [inline pytest_tests/tests/services/s3_gate/test_s3_bucket.py:34]: Whitespace. Is there any `go fmt` or `dotnet format` for Python? Can we run it once for our codebase, store a config for it and _never_ have these hunks again?

Comment (roman-khimov) [reply pytest_tests/tests/services/s3_gate/test_s3_bucket.py:34]: ef04906beaba71ca8d6e633b01601456f0582f47 tried to do that, but looks like it wasn't sufficient.

File: README.md (hunk 1 above)
Comment (evgeniiz321) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (evgeniiz321) [reply README.md:75]: done

File: pytest_tests/lib/helpers/http_gate.py (hunk 3 above)
Comment (evgeniiz321) [reply pytest_tests/lib/helpers/http_gate.py:245]: it is from black formatter, all files I ran through it

File: pytest_tests/tests/services/s3_gate/test_s3_bucket.py (hunk 4 above)
Comment (evgeniiz321) [reply pytest_tests/tests/services/s3_gate/test_s3_bucket.py:34]: We have _black_ formatter, and I always use it, that's why I always have some not related to PR changes. For this particular PR I ran it for all files.

Comment (roman-khimov) [reply pytest_tests/tests/services/s3_gate/test_s3_bucket.py:34]: The way to do it:
 * run it once for _everything_
 * commit these whitespace changes only
 * go on from this base

The problem is mixing whitespace with real changes, it's a mess to review.

**Comment by evgeniiz321** [comment]
@roman-khimov was there any changes in the neofs-node related to placement policies' syntax? On the node from master I'm receiving following error:

```
RuntimeError: Command: neofs-cli --config /home/runner/work/neofs-testcases/neofs-testcases/neofs-testcases/wallet_config.yml container create --rpc-endpoint 'localhost:37455' --wallet '/home/runner/work/neofs-testcases/neofs-testcases/neofs-testcases/TemporaryDir/6d12e12e-db20-4160-ab08-7c848e3afc78.json' --basic-acl '0FBFBFFF' --await --policy 'REP 2 IN X CBF 1 SELECT 4 FROM * AS X'
Error:
return code: 1
output: could not build container nodes based on given placement policy, use --force option to skip this check: not enough nodes to SELECT from: 'X'
```

**Comment by roman-khimov** [comment]
Policies are exactly the same, the problem can be in the number of nodes present in the system (see netmap). "SELECT 4" needs four storage nodes to succeed.

**Comment by evgeniiz321** [comment]
@roman-khimov the problem is that nothing has been changed neither in tests nor in env. The only changed part is the new version of the node binary

**Comment by evgeniiz321** [comment]
@roman-khimov one of the nodes are excluded from the network map, I'm not sure why, since there are no tests that does that, and it seems to be random. In the logs of this node I see a lot of errors like these:

```
2024-03-07T15:47:32.224Z	debug	route/calls.go:160	could not close remote server writer	{"component": "container_estimations", "key": "03a58c63ca3c531868dec09ebfdd4b6f8289b236bec9f909a67797d36ecdefdab9", "error": "status: code = 1024 message = node outside the container"}
2024-03-07T15:47:32.226Z	info	cache/multi.go:212	invalidated cached clients to the node caused by the error	{"address group": ["/dns4/localhost/tcp/59921"], "error": "/dns4/localhost/tcp/59921 node client: status: code = 1024 message = node outside the container"}
2024-03-07T15:47:32.226Z	debug	route/calls.go:160	could not close remote server writer	{"component": "container_estimations", "key": "0268cdb62e2484426110e10336878deadaa0063435c84db2d93774c3171f6f2f14", "error": "status: code = 1024 message = node outside the container"}
2024-03-07T15:47:32.228Z	info	cache/multi.go:212	invalidated cached clients to the node caused by the error	{"address group": ["/dns4/localhost/tcp/42857"], "error": "/dns4/localhost/tcp/42857 node client: status: code = 1024 message = node outside the container"}
2024-03-07T15:47:32.228Z	debug	route/calls.go:160	could not close remote server writer	{"component": "container_estimations", "key": "02890a6d47519552b289f76f33cc6e07d17fc0deaf890ecf7becba8ad407ac81bb", "error": "status: code = 1024 message = node outside the container"}
2024-03-07T15:47:32.228Z	debug	controller/calls.go:126	local load announcement successfully finished	{"component": "container_estimations", "epoch": 63, "stage": "p2p"}
2024-03-07T15:47:32.228Z	debug	controller/calls.go:182	announcement successfully interrupted	{"component": "container_estimations", "epoch": 63, "stage": "p2p"}
2024-03-07T15:47:32.300Z	error	policer/check.go:111	could not build placement vector for object	{"component": "Object Policer", "cid": "Zd37AyX8CuADFHq6k3hoYZTZghf8FmhVsmzFWChtDQV", "error": "could not get container nodes: not enough nodes to SELECT from: 'X'"}
```

And these:
```
2024-03-07T16:29:10.732Z	info	policer/check.go:151	node is outside the network map, holding the replica...	{"component": "Object Policer", "object": "5aAM9niS7EJeTta95D3kkGWW5TPDBuiMxpjJjzsph6pm/DPyCtCHZYvYdg4iLwUz9ZQX9asxtcNVVESnRs4PuB5vR"}
2024-03-07T16:29:10.733Z	error	replicator/process.go:62	could not replicate object	{"component": "Object Replicator", "node": "02890a6d47519552b289f76f33cc6e07d17fc0deaf890ecf7becba8ad407ac81bb", "object": "5aAM9niS7EJeTta95D3kkGWW5TPDBuiMxpjJjzsph6pm/8UpbtPqqGH3xZqhDeBJMfoxnvNpg3AaVqWtKiYHsNie9", "error": "(*putsvc.RemoteSender) could not send object: (*putsvc.remoteTarget) could not put object to [/dns4/localhost/tcp/42857]: write object payload into stream: invalid response signature: could not verify meta header: invalid public key"}
2024-03-07T16:29:10.733Z	error	replicator/process.go:62	could not replicate object	{"component": "Object Replicator", "node": "02890a6d47519552b289f76f33cc6e07d17fc0deaf890ecf7becba8ad407ac81bb", "object": "5aAM9niS7EJeTta95D3kkGWW5TPDBuiMxpjJjzsph6pm/H5xDH9r1yTAQMB3oqSMCdmUmxtAw2rQe8m7jD1HT5Tft", "error": "(*putsvc.RemoteSender) could not send object: (*putsvc.remoteTarget) could not put object to [/dns4/localhost/tcp/42857]: write object payload into stream: invalid response signature: could not verify meta header: invalid public key"}
```

Here is full logs - 
[neofs logs-6.zip](https://github.com/nspcc-dev/neofs-testcases/files/14528052/neofs.logs-6.zip) (The node that is excluded is - **sn_2_log**)

**Comment by roman-khimov** [comment]
> SubDiv: Stockholms lنn

Maybe related to LOCODE update in the recent node. Although it should't matter much if you have the same version of IR and SN.

**Review by roman-khimov — APPROVED** [review]
The last one like this. Whitespace must not be mixed with real changes.
